### ISSUE TITLE
dump benedict object to yaml not working in versions from 0.20 to 0.22

### ISSUE THREAD

**Issue by lburinov73** [fabiocaccamo/python-benedict#43]
**Python version**
3.8.0

**Package version**
0.20 - 0.22

**Current behavior (bug description)**
Code example:
```
#!/bin/python

from benedict import benedict
import yaml

data = benedict({"level1": {"level2": "blablabla"}})
data_yaml = yaml.safe_dump(dict(data))
print(data_yaml)
```
With `python-benedict==0.19.0`:
```
level1:
  level2: blablabla
```
With `python-benedict==0.22.0`:
```
Traceback (most recent call last):
  File "./test.py", line 7, in <module>
    data_yaml = yaml.safe_dump(dict(data))
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/__init__.py", line 306, in safe_dump
    return dump_all([data], stream, Dumper=SafeDumper, **kwds)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/__init__.py", line 278, in dump_all
    dumper.represent(data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 27, in represent
    node = self.represent_data(data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 48, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 207, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 118, in represent_mapping
    node_value = self.represent_data(item_value)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 58, in represent_data
    node = self.yaml_representers[None](self, data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 231, in represent_undefined
    raise RepresenterError("cannot represent an object", data)
yaml.representer.RepresenterError: ('cannot represent an object', {'level2': 'blablabla'})
```

`to_yaml()` not working too.
Code example:
```
#!/bin/python

from benedict import benedict

data = benedict({"level1": {"level2": "blablabla"}})
data_yaml = data.to_yaml()
print(data_yaml)
```
Result with `python-benedict==0.19.0`:
```
level1:
  level2: blablabla
```

Result with `python-benedict==0.22.0`:
```
level1: !!python/object/new:benedict.dicts.benedict
  dictitems:
    level2: blablabla
  state:
    _dict:
      level2: blablabla
    _keypath_separator: .
```

**Comment by fabiocaccamo** [comment]
@lburinov73 fixed in `0.22.1` version.

**Comment by lburinov73** [comment]
@fabiocaccamo Hi! `to_yaml()` is fixed, but something is wrong with dict compatibility

```
Python 3.8.0 (default, May  7 2020, 02:49:39) 
[GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from benedict import benedict
>>> import yaml
>>> yaml.safe_dump(dict(benedict({"level1": {"level2": "blablabla"}})))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/__init__.py", line 306, in safe_dump
    return dump_all([data], stream, Dumper=SafeDumper, **kwds)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/__init__.py", line 278, in dump_all
    dumper.represent(data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 27, in represent
    node = self.represent_data(data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 48, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 207, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 118, in represent_mapping
    node_value = self.represent_data(item_value)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 58, in represent_data
    node = self.yaml_representers[None](self, data)
  File "/home/lburinov/.local/lib/python3.8/site-packages/yaml/representer.py", line 231, in represent_undefined
    raise RepresenterError("cannot represent an object", data)
yaml.representer.RepresenterError: ('cannot represent an object', {'level2': 'blablabla'})
```

`python-benedict          0.22.1`

This code worked in 0.19...

**Comment by fabiocaccamo** [comment]
@lburinov73 fixed in `0.22.2` version.

**Comment by lburinov73** [comment]
@fabiocaccamo thanks!
